main.go

```diff
@@ -75,47 +113,94 @@ func (cmd *updateConfigCommand) run() error {
 		opt = helm.ReuseValues(true)
 	}
 
-	newValues, _ := yaml.Marshal(dest)
 	_, err = cmd.client.UpdateReleaseFromChart(
 		ls.Releases[0].Name,
 		ls.Releases[0].Chart,
-		helm.UpdateValueOverrides(newValues),
+		helm.UpdateValueOverrides(valBytes),
 		opt,
 	)
 
 	if err != nil {
-		return fmt.Errorf("Error: failed to update release", err)
+		return errors.Wrapf(err, "Failed to update release")
 	}
 
 	fmt.Printf("Info: update successfully\n")
 	return nil
 }
 
-func mergeValues(dest map[interface{}]interface{}, src map[string]interface{}) map[interface{}]interface{} {
+// mergeValues merges destination and source map, preferring values from the source map
+func mergeValues(dest map[string]interface{}, src map[string]interface{}) map[string]interface{} {
 	for k, v := range src {
 		// If the key doesn't exist, then just set the key to that value
 		if _, exists := dest[k]; !exists {
 			dest[k] = v
 			continue
 		}
 
-		nextMap, ok := v.(map[string]interface{})
+		nextMap, ok := v.(map[interface{}]interface{})
 		// If it isn't another map, overwrite the value
 		if !ok {
 			dest[k] = v
 			continue
 		}
 
+		// Convert nextMap key as string
+		convertedNextMap := map[string]interface{}{}
+		for k,v := range nextMap {
+			convertedNextMap[k.(string)] = v
+		}
+
 		// Edge case: If the key exists in the destination, but isn't a map
 		destMap, isMap := dest[k].(map[interface{}]interface{})
 		// If the source map has a map for this key, prefer it
 		if !isMap {
 			dest[k] = v
 			continue
 		}
-		// If they are bosh map, merge them
-		dest[k] = mergeValues(destMap, nextMap)
+
+		// Convert destMap key as string
+		convertedDestMap := map[string]interface{}{}
+		for k,v := range destMap{
+			convertedDestMap[k.(string)] = v
+		}
+
```

Comment:
Line 148-152 and Line 162-165 are basically the same, could you please change to use a function?